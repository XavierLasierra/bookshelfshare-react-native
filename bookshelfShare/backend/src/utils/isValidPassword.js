const bcrypt = require("bcrypt");

function isValidPassword(password) {
  return bcrypt.compareSync(password, this.password);
}

module.exports = { isValidPassword };
