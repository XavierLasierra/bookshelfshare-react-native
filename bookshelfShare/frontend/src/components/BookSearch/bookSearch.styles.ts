import { StyleSheet } from 'react-native';
import stylesConstants from '../../styles/styles.constants';

export default StyleSheet.create({
  searchContainer: {
    padding: stylesConstants.padding.normal,
    alignItems: 'center'
  },
  toggleContainer: {
    flexDirection: 'row',
    marginTop: 10
  },
  toggle: {
    alignItems: 'center',
    paddingVertical: 15,
    width: 150,
    backgroundColor: stylesConstants.colors.white
  },
  toggleActive: {
    backgroundColor: stylesConstants.colors.secondary
  },
  toggleText: {
    fontSize: stylesConstants.fontSize.big
  },
  formContainer: {
    width: '100%',
    height: 400,
    justifyContent: 'space-between'
  },
  searchButton: {
    marginTop: 20
  }
});
