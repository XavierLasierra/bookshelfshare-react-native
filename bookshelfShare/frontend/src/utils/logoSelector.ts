import BookIcon from '../assets/bookIcon.svg';
import SearchIcon from '../assets/searchIcon.svg';
import ProfileIcon from '../assets/profileIcon.svg';

export default function logoSelector(logoName: string) {
  switch (logoName) {
    case 'SearchIcon':
      return SearchIcon;
    case 'BookIcon':
      return BookIcon;
    case 'ProfileIcon':
      return ProfileIcon;
    default:
      return BookIcon;
  }
}
