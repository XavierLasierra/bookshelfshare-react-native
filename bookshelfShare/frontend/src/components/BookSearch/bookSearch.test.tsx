import React from 'react';
import BookSearch from './BookSearch';
import { render, fireEvent } from '../../utils/test.utils';

import { searchBooks } from '../../redux/actions/books.creator';
import booksActions from '../../redux/actions/books.actions';
import bookDetailsMock from '../../mocks/bookDetails.mock';

jest.mock('../../redux/actions/books.creator', () => ({
  searchBooks: jest.fn()
}));

const navigation = {
  push: jest.fn()
};

describe('Given a BookSearch component', () => {
  describe('When it is rendered', () => {
    describe('And isbnFromCamera is empty', () => {
      let screen: any;
      beforeEach(() => {
        (searchBooks as jest.Mock).mockReturnValue({
          type: booksActions.LOAD_BOOKS,
          data: bookDetailsMock
        });
        screen = render(<BookSearch navigation={navigation} isbnFromCamera="" />);
      });
      describe('And you stay on ISBN page', () => {
        test('Then should match the snapshot', () => {
          expect(screen).toMatchSnapshot();
        });

        describe('And you type on isbnInput', () => {
          test('Then the text typed should be rendered', () => {
            const isbnChange = '12345';
            const isbnInput = screen.getByTestId('isbnInput');
            fireEvent.changeText(isbnInput, isbnChange);

            expect(screen.queryByDisplayValue(isbnChange)).not.toBe(null);
          });
        });

        describe('And you click on serchButton', () => {
          describe('And isbn is not empty', () => {
            const isbnChange = '12345';
            beforeEach(() => {
              const isbnInput = screen.getByTestId('isbnInput');
              const searchButton = screen.getByTestId('searchButton');
              fireEvent.changeText(isbnInput, isbnChange);
              fireEvent.press(searchButton);
            });

            test('Then searchBooks should be called', () => {
              expect(searchBooks).toHaveBeenCalled();
            });

            test('Then navigation.push should have been called with BookResults and an object containing the isbn changed', () => {
              expect(navigation.push).toHaveBeenCalledWith('BookResults', {
                searchInformation: {
                  isbn: isbnChange,
                  inauthor: '',
                  intitle: '',
                  inpublisher: ''
                }
              });
            });
          });
          describe('And isbn is empty', () => {
            beforeEach(() => {
              const searchButton = screen.getByTestId('searchButton');
              fireEvent.press(searchButton);
            });

            test('Then searchBooks should have not been called', () => {
              expect(searchBooks).toHaveBeenCalled();
            });
          });
        });

        describe('And you click on isbnPageButton', () => {
          test('Then screen should not have changed', () => {
            const isbnPageButton = screen.getByTestId('isbnPageButton');
            fireEvent.press(isbnPageButton);
            expect(screen).toMatchSnapshot();
          });
        });
      });

      describe('And you change to Other page clicking otherPageButton', () => {
        beforeEach(() => {
          const otherPageButton = screen.getByTestId('otherPageButton');
          fireEvent.press(otherPageButton);
        });

        test('Then should match the snapshot', () => {
          expect(screen).toMatchSnapshot();
        });

        describe('And you type on authorInput', () => {
          test('Then the text typed should be rendered', () => {
            const authorChange = 'author';
            const authorInput = screen.getByTestId('authorInput');
            fireEvent.changeText(authorInput, authorChange);

            expect(screen.queryByDisplayValue(authorChange)).not.toBe(null);
          });
        });

        describe('And you type on titleInput', () => {
          test('Then the text typed should be rendered', () => {
            const titleChange = 'title';
            const titleInput = screen.getByTestId('titleInput');
            fireEvent.changeText(titleInput, titleChange);

            expect(screen.queryByDisplayValue(titleChange)).not.toBe(null);
          });
        });

        describe('And you type on publisherInput', () => {
          test('Then the text typed should be rendered', () => {
            const publisherChange = 'publisher';
            const publisherInput = screen.getByTestId('publisherInput');
            fireEvent.changeText(publisherInput, publisherChange);

            expect(screen.queryByDisplayValue(publisherChange)).not.toBe(null);
          });
        });

        describe('And you click on serchButton', () => {
          describe('And publisher is not empty', () => {
            const publisherChange = 'publisher';
            beforeEach(() => {
              const publisherInput = screen.getByTestId('publisherInput');
              const searchButton = screen.getByTestId('searchButton');
              fireEvent.changeText(publisherInput, publisherChange);
              fireEvent.press(searchButton);
            });

            test('Then searchBooks should be called', () => {
              expect(searchBooks).toHaveBeenCalled();
            });

            test('Then navigation.push should have been called with BookResults and an object containing the publisher typed', () => {
              expect(navigation.push).toHaveBeenCalledWith('BookResults', {
                searchInformation: {
                  isbn: '',
                  inauthor: '',
                  intitle: '',
                  inpublisher: publisherChange
                }
              });
            });
          });
        });

        describe('And you click on otherPageButton', () => {
          test('Then screen should not have changed', () => {
            const otherPageButton = screen.getByTestId('otherPageButton');
            fireEvent.press(otherPageButton);
            expect(screen).toMatchSnapshot();
          });
        });

        describe('And you click IsbnPageButton', () => {
          test('Then should match the snapshot', () => {
            const isbnPageButton = screen.getByTestId('isbnPageButton');
            fireEvent.press(isbnPageButton);
            expect(screen).toMatchSnapshot();
          });
        });
      });
    });

    describe('And isbnFromCamera has an isbn', () => {
      const isbnFromCamera = '1234';
      beforeEach(() => {
        (searchBooks as jest.Mock).mockReturnValue({
          type: booksActions.LOAD_BOOKS,
          data: bookDetailsMock
        });
        render(<BookSearch navigation={navigation} isbnFromCamera={isbnFromCamera} />);
      });
      test('Then searchBooks should be called', () => {
        expect(searchBooks).toHaveBeenCalled();
      });

      test('Then navigation.push should have been called with BookResults and an object containing the isbnFromCamera', () => {
        expect(navigation.push).toHaveBeenCalledWith('BookResults', {
          searchInformation: {
            isbn: isbnFromCamera,
            inauthor: '',
            intitle: '',
            inpublisher: ''
          }
        });
      });
    });
  });
});
