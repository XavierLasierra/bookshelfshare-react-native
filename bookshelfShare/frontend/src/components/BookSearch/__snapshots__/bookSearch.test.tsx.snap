// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a BookSearch component When it is rendered And isbnFromCamera is empty And you change to Other page clicking otherPageButton And you click IsbnPageButton Then should match the snapshot 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "#F5F5F5",
      "minHeight": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#F5F5F5",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D9D8C4",
          "borderBottomLeftRadius": 25,
          "justifyContent": "center",
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <IconMock
        height={45}
        width={45}
      />
    </View>
    <View
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#D9D8C4",
            "height": 20,
            "width": 20,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopRightRadius": 25,
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "padding": 30,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3E3C37",
          "fontSize": 35,
          "fontWeight": "500",
        }
      }
    >
      Book Search
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 10,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ECECEC",
            "opacity": 1,
            "paddingVertical": 15,
            "width": 150,
          }
        }
        testID="isbnPageButton"
      >
        <Text
          style={
            Object {
              "fontSize": 20,
            }
          }
        >
          ISBN
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "opacity": 1,
            "paddingVertical": 15,
            "width": 150,
          }
        }
        testID="otherPageButton"
      >
        <Text
          style={
            Object {
              "fontSize": 20,
            }
          }
        >
          Other
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "height": 400,
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "height": 70,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 12,
              "fontWeight": "bold",
              "transform": Array [
                Object {
                  "translateX": 5,
                },
                Object {
                  "translateY": 20,
                },
              ],
              "zIndex": 1,
            }
          }
        >
          ISBN
        </Text>
        <TextInput
          onChangeText={[Function]}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#ECECEC",
              "borderRadius": 10,
              "borderWidth": 1,
              "fontSize": 20,
              "height": "100%",
              "paddingHorizontal": 30,
              "paddingTop": 30,
            }
          }
          testID="isbnInput"
          value=""
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#000000",
            "borderRadius": 10,
            "display": "flex",
            "height": 70,
            "justifyContent": "center",
            "marginTop": 20,
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="searchButton"
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 20,
              "fontWeight": "500",
            }
          }
        >
          Search
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`Given a BookSearch component When it is rendered And isbnFromCamera is empty And you change to Other page clicking otherPageButton And you click on otherPageButton Then screen should not have changed 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "#F5F5F5",
      "minHeight": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#F5F5F5",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D9D8C4",
          "borderBottomLeftRadius": 25,
          "justifyContent": "center",
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <IconMock
        height={45}
        width={45}
      />
    </View>
    <View
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#D9D8C4",
            "height": 20,
            "width": 20,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopRightRadius": 25,
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "padding": 30,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3E3C37",
          "fontSize": 35,
          "fontWeight": "500",
        }
      }
    >
      Book Search
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 10,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "opacity": 1,
            "paddingVertical": 15,
            "width": 150,
          }
        }
        testID="isbnPageButton"
      >
        <Text
          style={
            Object {
              "fontSize": 20,
            }
          }
        >
          ISBN
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ECECEC",
            "opacity": 1,
            "paddingVertical": 15,
            "width": 150,
          }
        }
        testID="otherPageButton"
      >
        <Text
          style={
            Object {
              "fontSize": 20,
            }
          }
        >
          Other
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "height": 400,
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "height": 70,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 12,
              "fontWeight": "bold",
              "transform": Array [
                Object {
                  "translateX": 5,
                },
                Object {
                  "translateY": 20,
                },
              ],
              "zIndex": 1,
            }
          }
        >
          author
        </Text>
        <TextInput
          onChangeText={[Function]}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#ECECEC",
              "borderRadius": 10,
              "borderWidth": 1,
              "fontSize": 20,
              "height": "100%",
              "paddingHorizontal": 30,
              "paddingTop": 30,
            }
          }
          testID="authorInput"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "height": 70,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 12,
              "fontWeight": "bold",
              "transform": Array [
                Object {
                  "translateX": 5,
                },
                Object {
                  "translateY": 20,
                },
              ],
              "zIndex": 1,
            }
          }
        >
          title
        </Text>
        <TextInput
          onChangeText={[Function]}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#ECECEC",
              "borderRadius": 10,
              "borderWidth": 1,
              "fontSize": 20,
              "height": "100%",
              "paddingHorizontal": 30,
              "paddingTop": 30,
            }
          }
          testID="titleInput"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "height": 70,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 12,
              "fontWeight": "bold",
              "transform": Array [
                Object {
                  "translateX": 5,
                },
                Object {
                  "translateY": 20,
                },
              ],
              "zIndex": 1,
            }
          }
        >
          publisher
        </Text>
        <TextInput
          onChangeText={[Function]}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#ECECEC",
              "borderRadius": 10,
              "borderWidth": 1,
              "fontSize": 20,
              "height": "100%",
              "paddingHorizontal": 30,
              "paddingTop": 30,
            }
          }
          testID="publisherInput"
          value=""
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#000000",
            "borderRadius": 10,
            "display": "flex",
            "height": 70,
            "justifyContent": "center",
            "marginTop": 20,
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="searchButton"
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 20,
              "fontWeight": "500",
            }
          }
        >
          Search
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`Given a BookSearch component When it is rendered And isbnFromCamera is empty And you change to Other page clicking otherPageButton Then should match the snapshot 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "#F5F5F5",
      "minHeight": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#F5F5F5",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D9D8C4",
          "borderBottomLeftRadius": 25,
          "justifyContent": "center",
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <IconMock
        height={45}
        width={45}
      />
    </View>
    <View
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#D9D8C4",
            "height": 20,
            "width": 20,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopRightRadius": 25,
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "padding": 30,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3E3C37",
          "fontSize": 35,
          "fontWeight": "500",
        }
      }
    >
      Book Search
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 10,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "opacity": 1,
            "paddingVertical": 15,
            "width": 150,
          }
        }
        testID="isbnPageButton"
      >
        <Text
          style={
            Object {
              "fontSize": 20,
            }
          }
        >
          ISBN
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ECECEC",
            "opacity": 1,
            "paddingVertical": 15,
            "width": 150,
          }
        }
        testID="otherPageButton"
      >
        <Text
          style={
            Object {
              "fontSize": 20,
            }
          }
        >
          Other
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "height": 400,
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "height": 70,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 12,
              "fontWeight": "bold",
              "transform": Array [
                Object {
                  "translateX": 5,
                },
                Object {
                  "translateY": 20,
                },
              ],
              "zIndex": 1,
            }
          }
        >
          author
        </Text>
        <TextInput
          onChangeText={[Function]}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#ECECEC",
              "borderRadius": 10,
              "borderWidth": 1,
              "fontSize": 20,
              "height": "100%",
              "paddingHorizontal": 30,
              "paddingTop": 30,
            }
          }
          testID="authorInput"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "height": 70,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 12,
              "fontWeight": "bold",
              "transform": Array [
                Object {
                  "translateX": 5,
                },
                Object {
                  "translateY": 20,
                },
              ],
              "zIndex": 1,
            }
          }
        >
          title
        </Text>
        <TextInput
          onChangeText={[Function]}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#ECECEC",
              "borderRadius": 10,
              "borderWidth": 1,
              "fontSize": 20,
              "height": "100%",
              "paddingHorizontal": 30,
              "paddingTop": 30,
            }
          }
          testID="titleInput"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "height": 70,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 12,
              "fontWeight": "bold",
              "transform": Array [
                Object {
                  "translateX": 5,
                },
                Object {
                  "translateY": 20,
                },
              ],
              "zIndex": 1,
            }
          }
        >
          publisher
        </Text>
        <TextInput
          onChangeText={[Function]}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#ECECEC",
              "borderRadius": 10,
              "borderWidth": 1,
              "fontSize": 20,
              "height": "100%",
              "paddingHorizontal": 30,
              "paddingTop": 30,
            }
          }
          testID="publisherInput"
          value=""
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#000000",
            "borderRadius": 10,
            "display": "flex",
            "height": 70,
            "justifyContent": "center",
            "marginTop": 20,
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="searchButton"
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 20,
              "fontWeight": "500",
            }
          }
        >
          Search
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`Given a BookSearch component When it is rendered And isbnFromCamera is empty And you stay on ISBN page And you click on isbnPageButton while on isbnPage Then screen should not have changed 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "#F5F5F5",
      "minHeight": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#F5F5F5",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D9D8C4",
          "borderBottomLeftRadius": 25,
          "justifyContent": "center",
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <IconMock
        height={45}
        width={45}
      />
    </View>
    <View
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#D9D8C4",
            "height": 20,
            "width": 20,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopRightRadius": 25,
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "padding": 30,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3E3C37",
          "fontSize": 35,
          "fontWeight": "500",
        }
      }
    >
      Book Search
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 10,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ECECEC",
            "opacity": 1,
            "paddingVertical": 15,
            "width": 150,
          }
        }
        testID="isbnPageButton"
      >
        <Text
          style={
            Object {
              "fontSize": 20,
            }
          }
        >
          ISBN
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "opacity": 1,
            "paddingVertical": 15,
            "width": 150,
          }
        }
        testID="otherPageButton"
      >
        <Text
          style={
            Object {
              "fontSize": 20,
            }
          }
        >
          Other
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "height": 400,
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "height": 70,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 12,
              "fontWeight": "bold",
              "transform": Array [
                Object {
                  "translateX": 5,
                },
                Object {
                  "translateY": 20,
                },
              ],
              "zIndex": 1,
            }
          }
        >
          ISBN
        </Text>
        <TextInput
          onChangeText={[Function]}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#ECECEC",
              "borderRadius": 10,
              "borderWidth": 1,
              "fontSize": 20,
              "height": "100%",
              "paddingHorizontal": 30,
              "paddingTop": 30,
            }
          }
          testID="isbnInput"
          value=""
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#000000",
            "borderRadius": 10,
            "display": "flex",
            "height": 70,
            "justifyContent": "center",
            "marginTop": 20,
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="searchButton"
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 20,
              "fontWeight": "500",
            }
          }
        >
          Search
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`Given a BookSearch component When it is rendered And isbnFromCamera is empty And you stay on ISBN page Then should match the snapshot 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "#F5F5F5",
      "minHeight": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#F5F5F5",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D9D8C4",
          "borderBottomLeftRadius": 25,
          "justifyContent": "center",
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <IconMock
        height={45}
        width={45}
      />
    </View>
    <View
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#D9D8C4",
            "height": 20,
            "width": 20,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopRightRadius": 25,
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "padding": 30,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3E3C37",
          "fontSize": 35,
          "fontWeight": "500",
        }
      }
    >
      Book Search
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 10,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ECECEC",
            "opacity": 1,
            "paddingVertical": 15,
            "width": 150,
          }
        }
        testID="isbnPageButton"
      >
        <Text
          style={
            Object {
              "fontSize": 20,
            }
          }
        >
          ISBN
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "opacity": 1,
            "paddingVertical": 15,
            "width": 150,
          }
        }
        testID="otherPageButton"
      >
        <Text
          style={
            Object {
              "fontSize": 20,
            }
          }
        >
          Other
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "height": 400,
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "height": 70,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 12,
              "fontWeight": "bold",
              "transform": Array [
                Object {
                  "translateX": 5,
                },
                Object {
                  "translateY": 20,
                },
              ],
              "zIndex": 1,
            }
          }
        >
          ISBN
        </Text>
        <TextInput
          onChangeText={[Function]}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#ECECEC",
              "borderRadius": 10,
              "borderWidth": 1,
              "fontSize": 20,
              "height": "100%",
              "paddingHorizontal": 30,
              "paddingTop": 30,
            }
          }
          testID="isbnInput"
          value=""
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#000000",
            "borderRadius": 10,
            "display": "flex",
            "height": 70,
            "justifyContent": "center",
            "marginTop": 20,
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="searchButton"
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 20,
              "fontWeight": "500",
            }
          }
        >
          Search
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
